### PR TITLE
Clean up stale/broken JavaScript and remove jQuery

### PR DESCRIPTION
**Pull request by doeg**
This fixes the errors getting thrown by the [tocify plugin](http://gregfranko.com/jquery.tocify.js/):

image

...and by "fixes", I mean "delete a bunch of code". :) 

On investigation, it turns out we can completely remove jQuery and save ~115 KB of production assets. 🎉 Anything that's currently "using" it is broken or unnecessary. I've added some inline explanations.

### CODE REVIEW

File: assets/js/app.js [hunk 1]
```diff
@@ -1,72 +1,7 @@
...
-}
-
-function showAndHideTitle() {
-  if (elementExists('.docs-article')) {
-    
-  }
-}
-
-function tableOfContents(options) {
-  $('#tableOfContents').tocify(options);
```
Comment (doeg) [inline assets/js/app.js:63]: [tocify](http://gregfranko.com/jquery.tocify.js/) is intended to highlight the headings in a table of contents nav element when scrolling. 

I think this is a nice bit of functionality for sure! But it's been broken for a long time and has a big jQuery dependency, hence simply removing it for now.

If/when we do want to reintroduce this feature, it's fairly straightforward to do this with plain JavaScript; here's a couple of examples: [one](https://css-tricks.com/sticky-table-of-contents-with-scrolling-active-states/)), [two](https://lab.hakim.se/progress-nav/).

File: assets/js/app.js [hunk 2]
```diff
@@ -1,72 +1,7 @@
...
-  const navbarHeight = $('.navbar').height();
-  const extraPadding = 20;
-  const navbarOffset = -1 * (navbarHeight + extraPadding);
-  var shiftWindow = function() { scrollBy(0, navbarOffset) };
-  window.addEventListener("hashchange", shiftWindow);
-  window.addEventListener("pageshow", shiftWindow);
-  function load() { if (window.location.hash) shiftWindow(); }
-}
-
-function fixUponScroll() {
```
Comment (doeg) [inline assets/js/app.js:34]: I think this is intended to hide/show the fixed navbar when scrolling. Another nice feature, but it's broken as written.

File: assets/js/app.js [hunk 3]
```diff
@@ -1,72 +1,7 @@
...
-    });
+window.addEventListener('DOMContentLoaded', () => {
+  const burger = document.getElementById('navbar-burger');
+  burger.addEventListener('click', () => {
+    const menu = document.getElementById('navbar-menu');
+    [burger, menu].forEach(el => el.classList.toggle('is-active'));
   });
-}
-
-function linkClickOffset() {
```
Comment (doeg) [inline assets/js/app.js:24]: This bit of code has not been getting called due to the prior error thrown by the `fixUponScroll` function, and in any case is obviated by equivalent functionality added by https://github.com/vitessio/website/pull/617.

File: assets/js/app.js [hunk 4]
```diff
@@ -1,72 +1,7 @@
...
-  showAndHide: false,
-  smoothScroll: true,
-  scrollTo: $('.navbar').height() + 25
-}
-
-function elementExists(selector) {
-  return $(selector).length > 0;
-}
-
-function navbarBurgerToggle() {
```
Comment (doeg) [inline assets/js/app.js:13]: This is the only bit of JavaScript we need to keep; it's been rewritten without jQuery.

**Review by askdba — APPROVED** [review]
LGTM

**Pull request merged**